PocketSocket/PSWebSocket.m

```diff
@@ -160,14 +160,26 @@ - (instancetype)initClientSocketWithRequest:(NSURLRequest *)request {
         if(port == 0) {
             port = (_secure) ? 443 : 80;
         }
-        
+
         CFReadStreamRef readStream = nil;
         CFWriteStreamRef writeStream = nil;
         CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault,
                                            (__bridge CFStringRef)host,
                                            port,
                                            &readStream,
                                            &writeStream);
+
+        CFDictionaryRef systemProxyConfig = CFNetworkCopySystemProxySettings();
+        if(systemProxyConfig != nil) {
+            // XXX: kCFNetworkProxiesSOCKSEnable is not available on iOS
+            NSNumber *socksEnabled = CFDictionaryGetValue(systemProxyConfig, @"SOCKSEnable");
+            if(socksEnabled != nil && [socksEnabled boolValue]) {
+                CFReadStreamSetProperty(readStream, kCFStreamPropertySOCKSProxy, systemProxyConfig);
+                CFWriteStreamSetProperty(writeStream, kCFStreamPropertySOCKSProxy, systemProxyConfig);
+            }
+            CFRelease(systemProxyConfig);
+        }
+
```

Comment:
Instead of using the raw string above, could we also wrap this whole block of code into a conditional compilation clause like this:

```
#if TARGET_OS_OSX
/* new code... */
#endif
```

and use the actual key instead? Or do we actually have to do this socks configuration on iOS, even though the key is not supported there?

Reply:
Thx. I think we can constrain it to os_osx only. As far as I know, iOS does not support socks proxy.